### PR TITLE
#6 Task. Alexander Zenchyk

### PR DESCRIPTION
**Pull request by zeeenjaaa**
https://travis-ci.com/github/zeeenjaaa/js-assignments/builds/175584061

### CODE REVIEW

File: task/06-conditions-n-loops-tasks.js [hunk 1]
```diff
@@ -30,7 +30,7 @@
  *
  */
 function getFizzBuzz(num) {
-    throw new Error('Not implemented');
+ return num % 15 == 0 ? 'FizzBuzz': num % 3 == 0 ? 'Fizz': num % 5 == 0 ?'Buzz': num;
```
Comment (vramaniuk) [inline task/06-conditions-n-loops-tasks.js]: "operator === is always preferable
sometimes ''==' may cause an error
(check at console :
""0"" == 0
and ""0"" === 0)"

File: task/06-conditions-n-loops-tasks.js [hunk 2]
```diff
@@ -166,7 +176,16 @@ function isInsideCircle(circle, point) {
  *   'entente' => null
  */
 function findFirstSingleChar(str) {
-    throw new Error('Not implemented');
+   // throw new Error('Not implemented');
+    let ans;
+    for (let i = 0 ; i < str.length; i++){
+        if( str.lastIndexOf(str[i]) == str.indexOf(str[i])){
```
Comment (vramaniuk) [inline task/06-conditions-n-loops-tasks.js]: "operator === is always preferable

File: task/06-conditions-n-loops-tasks.js [hunk 3]
```diff
@@ -192,7 +211,7 @@ function findFirstSingleChar(str) {
  *
  */
 function getIntervalString(a, b, isStartIncluded, isEndIncluded) {
-    throw new Error('Not implemented');
+  return `${isStartIncluded?'[':'('}${a>b?b:a}, ${a>b?a:b}${isEndIncluded?']':')'}`
```
Comment (vramaniuk) [inline task/06-conditions-n-loops-tasks.js]: Please try to follow these rules 
https://javascript.info/coding-style

File: task/06-conditions-n-loops-tasks.js [hunk 4]
```diff
@@ -372,8 +428,27 @@ function toNaryString(num, n) {
...
+  let ans =[];
+  let firstString = pathes[0].split('/');
+  let counter = 1;  
+
+  for (let i = 0; i < firstString.length; i++){
+
+    for (let j =1; j< pathes.length; j++){
+
+      let otherString = pathes[j].split('/');  
+      firstString[i]==otherString[i]? counter++:counter=1;
```
Comment (vramaniuk) [inline task/06-conditions-n-loops-tasks.js]: operator === is always preferable

File: task/06-conditions-n-loops-tasks.js [hunk 5]
```diff
@@ -372,8 +428,27 @@ function toNaryString(num, n) {
...
+      if(counter==pathes.length){
```
Comment (vramaniuk) [inline task/06-conditions-n-loops-tasks.js]: operator === is always preferable

File: task/06-conditions-n-loops-tasks.js [hunk 6]
```diff
@@ -431,10 +517,29 @@ function getMatrixProduct(m1, m2) {
  *
  */
 function evaluateTicTacToePosition(position) {
-    throw new Error('Not implemented');
+   let tmpArr = position;
+   let wins  =[[0,1,2],[3,4,5],[6,7,8],[0,4,8],[2,4,6],[0,3,6],[1,4,7],[2,5,8]];
+    
+   for (let i = 0; i < 3; i++){
+    for (let j = 0; j < 3; j++){
+      if( tmpArr[i][j] != 'X' && tmpArr[i][j] != '0'){
```
Comment (vramaniuk) [inline task/06-conditions-n-loops-tasks.js]: "operator !== is always preferable
sometimes ''!=' may cause an error
(check at console :
""0"" != 0
and ""0"" !== 0)"

File: task/06-conditions-n-loops-tasks.js [hunk 7]
```diff
@@ -431,10 +517,29 @@ function getMatrixProduct(m1, m2) {
...
+      }
+    }
+   }
+  tmpArr= tmpArr.flat();
+  wins = wins.flat(); 
+   for (let i = 0; i < wins.length; i=i+3){
+    if(tmpArr[wins[i]] =='X' && tmpArr[wins[i+1]] == 'X' && tmpArr[wins[i+2]] == 'X'){
+      return 'X'
+    }
+    if(tmpArr[wins[i]] =='0' && tmpArr[wins[i+1]] == '0' && tmpArr[wins[i+2]] == '0'){
```
Comment (vramaniuk) [inline task/06-conditions-n-loops-tasks.js]: operator === is always preferable

**Review by vramaniuk — COMMENTED** [review]
I'll mark the PR as "Draft", please click "ready for review" when it will be finished. 
Please, follow these rules https://javascript.info/coding-style
Thank you!

**Comment by vramaniuk** [comment]
Resolove conflicting files (readme.md)

**Comment by zeeenjaaa** [comment]
Thank you for review and good lesson and time! 
I've fixed all issues except 'readme.md' because in first steps of this course it was needed to change that file.

**Comment by vramaniuk** [comment]
image
Please, look what's wrong
https://travis-ci.org/github/AisBrestEDU/js-assignments/builds/717895136

**Comment by vramaniuk** [comment]
Closing the PR as accepted! Thanks!

**Comment by zeeenjaaa** [comment]
As I see tests in AIS repositry have been updated. 
So I merged changes and update npm. All now  is fine. Thank you!
https://travis-ci.com/github/zeeenjaaa/js-assignments/builds/179736397

**Pull request closed without merging**